src/vellum/workflows/workflows/base.py

```diff
@@ -196,6 +197,40 @@ def get_nodes(cls) -> Iterator[Type[BaseNode]]:
                     nodes.add(node)
                     yield node
 
+    @classmethod
+    def get_nodes_not_in_graph(cls) -> Iterator[Type[BaseNode]]:
+        """
+        Returns an iterator over the nodes that are defined but not used in the graph.
+        """
+        if not hasattr(cls, "unused"):
+            yield from ()
+        else:
+            nodes = set()
+            for item in cls.unused:
+                if isinstance(item, Graph):
+                    # Item is a graph
+                    for node in item.nodes:
+                        if node not in nodes:
+                            nodes.add(node)
+                            yield node
+                elif isinstance(item, set):
+                    # Item is a set of graphs or nodes
+                    for subitem in item:
+                        if isinstance(subitem, Graph):
+                            for node in subitem.nodes:
+                                if node not in nodes:
+                                    nodes.add(node)
+                                    yield node
+                        elif issubclass(subitem, BaseNode):
+                            if subitem not in nodes:
+                                nodes.add(subitem)
+                                yield subitem
+                elif issubclass(item, BaseNode):
+                    # Item is a node
+                    if item not in nodes:
+                        nodes.add(item)
+                        yield item
+
```

Comment:
[oos] Let's also add a `get_unused_edges(cls):` since serialization will need both. Can be deferred to the next PR